File: Practica11/A/Java/Main.java
public class Main {
    public static void main(String[] args) {
        Artista a1 = new Artista("Luis", "123", 10), a2 = new Artista("Ana", "456", 5);
        Artista a3 = new Artista("Mario", "789", 8), a4 = new Artista("Lucía", "321", 12);
        Anuncio an1 = new Anuncio(1, 1000.0);
        Pintura p1 = new Pintura("Atardecer", "óleo", a1, a2, an1, "paisaje");
        Pintura p2 = new Pintura("Aurora", "acrílico", a3, a4, null, "abstracto");
        System.out.println("Artistas registradoss:");
        Artista[] artistas = {a1, a2, a3, a4};
        for (Artista art : artistas)
            System.out.println("- " + art.nombre + " (CI: " + art.ci + ", Años: " + art.añosExperiencia + ")");
        Artista mayor = a1;
        for (Artista art : artistas)
            if (art.añosExperiencia > mayor.añosExperiencia) mayor = art;
        System.out.println("\nArtista con más experiencia: " + mayor.nombre);
        p2.anuncio = new Anuncio(2, 1500.0);
        double total = p1.anuncio.precio + p2.anuncio.precio;
        System.out.println("Monto total de venta: " + total);
    }
}

File: Practica11/A/Python/Main.py
from Artista import Artista
from Anuncio import Anuncio
from Pintura import Pintura
a1 = Artista("Luis", "123", 10)
a2 = Artista("Ana", "456", 5)
an1 = Anuncio(1, 1000.0)
p1 = Pintura("Atardecer", "óleo", a1, a2, an1, "paisaje")
a3 = Artista("Mario", "789", 8)
a4 = Artista("Lucía", "321", 12)
p2 = Pintura("Aurora", "acrílico", a3, a4, None, "abstracto")
print("Artistas registrados:")
for artista in [a1, a2, a3, a4]:
    print(f"- {artista.nombre} (CI: {artista.ci}, Años: {artista.añosExp})")
mayor = max([a1, a2, a3, a4], key=lambda x: x.añosExp)
print(f"\nArtista con más experiencia: {mayor.nombre}")
p2.anuncio = Anuncio(2, 1500.0)
total = p1.anuncio.precio + p2.anuncio.precio
print(f"\nMonto total de venta: {total}")
